synapse/metrics/background_process_metrics.py

```diff
@@ -238,3 +234,42 @@ def wrap_as_background_process_inner_2(*args, **kwargs):
         return wrap_as_background_process_inner_2
 
     return wrap_as_background_process_inner
+
+
+class BackgroundProcessLoggingContext(LoggingContext):
+    """A logging context that tracks in flight metrics for background
+    processes.
+    """
+
+    __slots__ = ["_proc"]
+
+    def __init__(self, name: str):
+        super().__init__(name)
+
+        self._proc = _BackgroundProcess(name, self)
+
+    def start(self, rusage: "Optional[resource._RUsage]"):
```

Comment:
So does this run on every batch, or just when the process stops (runs out of things to process) and then starts up again?

Reply:
The naming is a bit confusing here. Basically `LoggingContext.{__enter__,__exit__}` are called first time we enter the context and last time we leave the context, while `{start,stop}` get called whenever the logging context starts/stops being the active/current logging context. (The names `resume` and `pause` might have been better here)

Reply:
Ok, and "entering the context" occurs each time a background processes starts processing.

So `_background_processes_active_since_last_scrape` contains just those processes that have *become active* since the last scrape, versus those that are *continuously* active.

Reply:
Yup, exactly

Reply:
Could you add a quick comment to where `_background_process_in_flight_count` is instantiated mentioning this distinction please?